chrome/content/zotero/xpcom/itemTreeManager.js

```diff
@@ -0,0 +1,379 @@
+		}
+		this._addPluginShutdownObserver();
+		await this._notifyItemTrees();
+		return registeredDataKeys;
+	}
+
+	/**
+	 * Unregister a custom column.
+	 * Although it's async, resolving does not promise the item trees are updated.
+	 * @param {string | string[]} dataKeys - The dataKey of the column to unregister
+	 * @returns {boolean} true if the column(s) are unregistered
+	 * @example
+	 * ```js
+	 * Zotero.ItemTreeManager.unregisterColumns('rtitle');
+	 * ```
+	 */
+	async unregisterColumns(dataKeys) {
+		const success = this._removeColumns(dataKeys);
+		if (!success) {
+			return false;
+		}
+		await this._notifyItemTrees();
+		return true;
+	}
+
+	// TODO: add cell renderer registration
+
+	/**
+	 * Get column(s) that matches the properties of option
+	 * @param {undefined | Partial.<ItemTreeColumnOptions> | Partial.<ItemTreeColumnOptions>[]} options - An option or array of options to match
+	 * @returns {ItemTreeColumnOptions[]}
+	 */
+	getCustomColumns(options) {
+		const allColumns = this._getColumnsByType("custom");
+		if (!options) {
+			return allColumns;
+		}
+		if (!Array.isArray(options)) {
+			options = [options];
+		}
+		options.forEach(o => o.dataKey = this._namespacedDataKey(o));
+		/** @type {ItemTreeColumnOptions[]} */
+		const matches = [];
+		for (const opt of options) {
+			const currentMatches = allColumns
+				.filter(col =>
+					// Aoid dataKey collision
+					!matches.map(match => match.dataKey).includes(col.dataKey)
+					// Match all properties
+					&& Object.keys(opt).every(key => {
+						// Ignore undefined properties
+						if (opt[key] === undefined) {
+							return true;
+						}
+						// Special case for enabledTreeIDs
+						if (key === "enabledTreeIDs"){
+							// If enabledTreeIDs is "*", match all tree IDs
+							return opt.enabledTreeIDs.includes("*")
+							// Otherwise, match the tree IDs
+								|| opt.enabledTreeIDs.every(treeID => (col.enabledTreeIDs || ["main"]).includes(treeID));
+						}
+						return col[key] === opt[key];
+					})
+				);
+			matches.push(...currentMatches);
+		}
+		return matches;
+	}
+
+	/**
+	 * Check if a column is registered as a custom column
+	 * @param {string} dataKey 
+	 * @returns {boolean} true if the column is registered as a custom column
+	 */
+	isCustomColumn(dataKey) {
+		return !!this._customColumns[dataKey];
+	}
+
+	/**
+	 * A centralized data source for custom columns. This is used by the ItemTreeRow to get data.
+	 * @param {Zotero.Item} item - The item to get data from
+	 * @param {string} dataKey - The dataKey of the column
+	 * @returns {string}
+	 */
+	getCustomCellData(item, dataKey) {
+		const options = this._customColumns[dataKey];
+		if (options && options.dataProvider) {
+			return options.dataProvider(item, dataKey);
+		}
+		return "";
+	}
+
+	/**
+	 * Get columns by type. Only support itemtree and custom for now.
+	 * @param {"itemtree" | "custom" | "*"} type 
+	 * @returns {ItemTreeColumnOptions[]}
+	 */
+	_getColumnsByType(type) {
```

Comment:
This file should only be concerned with columns registered by plugins. There are only two types of column types, ones passed in into trees as props and ones registered by plugins, i.e. custom. I don't see this reasonably changing, and when/if it does we can make adjustments. This should just be dropped and simplified.

Reply:
I will remove this